### PR TITLE
[core] Fast return if rollback verion equals lastest version

### PR DESCRIPTION
**Pull request by askwang**
### Purpose

If the rollback snapshot version is equal to the lastest snapshot version, we should fast return.

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java [hunk 1]
```diff
@@ -625,14 +630,19 @@ public void rollbackTo(String tagName) {
         checkArgument(tagManager.tagExists(tagName), "Rollback tag '%s' doesn't exist.", tagName);
 
         Snapshot taggedSnapshot = tagManager.taggedSnapshot(tagName);
+        SnapshotManager snapshotManager = snapshotManager();
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java:633]: Add a blank line.

File: paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java [hunk 2]
```diff
@@ -512,6 +512,11 @@ public void rollbackTo(long snapshotId) {
                 "Rollback snapshot '%s' doesn't exist.",
                 snapshotId);
 
+        // fast return
+        if (new Long(snapshotId).equals(snapshotManager.latestSnapshotId())) {
```
Comment (wwj6591812) [inline paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java:516]: I think new Long has a little strange.
Why not:

Long latestSnapshotId = snapshotManager.latestSnapshotId();
if (latestSnapshotId == null) {
    exception or error log
}
if (taggedSnapshot.id() == latestSnapshotId) {
    fast return
}

**Comment by wwj6591812** [comment]
A question:
Assuming that when the user executes the rollback procedure with snapshot = 1, the latestSnapshot Id of the Paimon table is 1, and before the rollback procedure ends, the latestSnapshot Id of the Paimon table changes to 2. After you apply this patch, in the end, the rollback procedure fast return so that the result is not snapshot = 1.
 Is this the result that the user wants?

**Comment by askwang** [comment]
> A question: Assuming that when the user executes the rollback procedure with snapshot = 1, the latestSnapshot Id of the Paimon table is 1, and before the rollback procedure ends, the latestSnapshot Id of the Paimon table changes to 2. After you apply this patch, in the end, the rollback procedure fast return, so the RollbackHelper#cleanLargerThan will not be called. Is this the result that the user wants?

In this case, the result is not what the user wants. But although there does not fast return, this case may also occur. For example, the rollback snapshot is 1 and the latest snapshot of table is also 1, before rollback ends the lastest snapshot of table changes to 2 or 3, the rollback logic gets the lastest shotshot is diffcult to keep always the newest, this will cause some new snapshots not to be processed.

**Comment by JingsongLi** [comment]
Hi @askwang , what is this PR aims to? just for performance?

**Comment by JingsongLi** [comment]
> > Hi @askwang , what is this PR aims to? just for performance?
> 
> yes, juse fast return, no need to execute RollbackHelper#cleanLargerThan.

Do we need to care about the speed of this operation? I think it should be very fast even no optimization.

**Pull request closed without merging**